hcloud/client_helper.go

```diff
@@ -51,3 +56,28 @@ func firstByName[T any](name string, listFn func() ([]*T, *Response, error)) (*T
 
 	return firstBy(listFn)
 }
+
+// getByIDOrName fetches the resource by ID when the identifier is an integer, otherwise
+// by Name. To support resources that have a integer as Name, an additional attempt is
+// made to fetch the resource by Name using the ID.
+//
+// This function is only meaningful for resources created by users.
```

Comment:
```suggestion
// This function is only meaningful for resources created by users.
```

Can you explain what you mean by this? Is this meant as a recommendation to use the `ID` whereever possible?

Reply:
I rewrote the comment to make it more explicit.